### PR TITLE
Overhaul INDRA World documentation

### PR DESCRIPTION
**Pull request by bgyori**
This PR creates a new documentation structure for the package after implementing the new INDRA World service architecture, and porting multiple modules from INDRA.

### CODE REVIEW

File: indra_world/sources/sofia/processor.py [hunk 1]
```diff
@@ -107,12 +108,39 @@ def _build_influences(self, rel_dict):
...
 
     def get_event(self, event_entry):
+        """Get an Event with the pre-set grounding mode
+
+        The grounding mode is set at initialization of the class and is
+        stored in the attribute `grounding_mode`.
+
+        Parameters
+        ----------
+        event_entry : dict[str, str]
```
Comment (cthoyt) [inline indra_world/sources/sofia/processor.py:118]: would be a great time to introduce some type hints ;)

Comment (cthoyt) [reply indra_world/sources/sofia/processor.py:118]: Want me to make a commit with some of these added?

Comment (bgyori) [reply indra_world/sources/sofia/processor.py:118]: If you don't have anything more exciting in mind, sure! But not critical in this case, I think.

Comment (cthoyt) [reply indra_world/sources/sofia/processor.py]: Done in 569a349

Comment (cthoyt) [reply indra_world/sources/sofia/processor.py]: I was even able to steal the types @johnbachman wrote in the sklearn/belief PR too ;)

Comment (bgyori) [reply indra_world/sources/sofia/processor.py]: Great, thanks!

**Pull request merged**